src/Sinch/Fax/Services/ServicesClient.cs

```diff
@@ -0,0 +1,301 @@
+        ///     Auto List emails for a number.
+        /// </summary>
+        /// <param name="serviceId">The serviceId containing the numbers you want to list.</param>
+        /// <param name="phoneNumber">The phone number you want to get emails for.</param>
+        /// <param name="page">The page number to fetch. If not specified, the first page will be returned.</param>
+        /// <param name="pageSize">Number of items to return on each page.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A list of emails addresses</returns>
+        IAsyncEnumerable<string> ListEmailsForNumberAuto(string serviceId, string phoneNumber, int? page = 1,
+            int? pageSize = 1000,
+            CancellationToken cancellationToken = default);
+
+        /// <summary>
+        ///     List numbers for a service.
+        /// </summary>
+        /// <param name="serviceId"></param>
+        /// <param name="page"></param>
+        /// <param name="pageSize"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        Task<ListNumbersResponse> ListNumbers(string serviceId, int? page, int? pageSize,
+            CancellationToken cancellationToken = default);
+
+        /// <summary>
+        ///     Auto List numbers for a service.
+        /// </summary>
+        /// <param name="serviceId"></param>
+        /// <param name="page"></param>
+        /// <param name="pageSize"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        IAsyncEnumerable<ServicePhoneNumber> ListNumbersAuto(string serviceId, int? page, int? pageSize,
+            CancellationToken cancellationToken = default);
+    }
+
+    internal sealed class ServicesClient : ISinchFaxServices
+    {
+        private readonly string _projectId;
+        private readonly Uri _baseAddress;
+        private readonly ILoggerAdapter<ISinchFaxServices>? _logger;
+        private readonly IHttp _http;
+        private readonly Uri _apiBasePath;
+
+        public ServicesClient(string projectId, Uri baseAddress, ILoggerAdapter<ISinchFaxServices>? logger, IHttp http)
+        {
+            _projectId = projectId;
+            _baseAddress = baseAddress;
+            _logger = logger;
+            _http = http;
+            _apiBasePath = new Uri(_baseAddress, $"/v3/projects/{projectId}/services");
+        }
+
+        public Task<Service> Create(CreateServiceRequest request, CancellationToken cancellationToken = default)
+        {
+            _logger?.LogInformation("Creating a service for {projectId}", _projectId);
+
+            return _http.Send<CreateServiceRequest, Service>(_apiBasePath, HttpMethod.Post, request, cancellationToken);
+        }
+
+        public Task<Service> Get(string serviceId, CancellationToken cancellationToken = default)
+        {
+            _logger?.LogInformation("Getting a service with {id} for {project}", serviceId, _projectId);
```

Comment:
Nit: `project` is referenced by its ID, so I would be preferable to rename `project` with `projectId`